docs_3x.md

```diff
@@ -139,7 +139,13 @@ Note: since this component will re-render its children every time its parent's p
   There are 3 features that this library provides in order to leverage offline capabilities in your Redux store: a reducer, a middleware and an offline queue system. You can use all of them or just the ones that suits your needs.
 
   ### Network reducer
-A network reducer to be provided to the store.
+A network reducer to be provided to the store. The reducer can take an optional comparisonFn. If no function is provided default comparison function will be used.
```

Comment:
That introduction should be placed at the beginning of the section [here](https://github.com/rgommezz/react-native-offline#1--give-the-network-reducer-to-redux).

## Usage

The library exposes a simple reducer or a factory function to tailor it a bit more to your needs.

### Basic reducer
This reducer will use a preconfigured comparison function for the [offline queue](https://github.com/rgommezz/react-native-offline#offline-queue). Please check the docs [here we should link to a new section inside offline queue where we explain the default behaviour, which is `lodash.isEqual` for plain actions and `thunk.toString()` for thunks/functions](). We should basically explain there that __If a similar action already existed on the queue, we remove it and push it again to the end of the queue__

```js
// configureStore.js
import { createStore, combineReducers } from 'redux'
import { reducer as network } from 'react-native-offline';

const rootReducer = combineReducers({
  // ... your other reducers here ...
  network,
});

const store = createStore(rootReducer);
export default store;
```

### Factory function
This may be useful if you are using the [offline queue](https://github.com/rgommezz/react-native-offline#offline-queue) and would like to have more control on how it behaves, particularly when determining whether an action already exists on the queue. That's why we expose a factory function as well, so that you can provide your own comparison function. You'd use it in the next way:

```js
// configureStore.js
import { createStore, combineReducers } from 'redux'
import { reducer as network } from 'react-native-offline';
import { comparisonFn } from './utils';
const rootReducer = combineReducers({
  // ... your other reducers here ...
  network(comparisonFn),
});
const store = createStore(rootReducer);
export default store;
```

The comparison function receives the `action` dispatched when offline and the current `actionQueue`. The result of the function will be either `undefined` meaning no match found or the `action` that matches the passed in action.
```js
function comparisonFn(
  action: ReduxAction | ReduxThunk,
  actionQueue: Array<ReduxAction | ReduxThunk>,
): ?(ReduxAction | ReduxThunk)
```